### PR TITLE
Migrate to async

### PR DESCRIPTION
**Pull request by mikeminutillo**
Migrate promises code to async.

All changes smoke tested. Discovered https://github.com/Particular/ServicePulse/pull/1676 and fixed it on a separate PR.

### CODE REVIEW

File: src/ServicePulse.Host/vue/src/components/failedmessages/AllDeletedMessages.vue
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/AllDeletedMessages.vue:1]: `loadMessages` does not seem to be awaited anywhere, is this right?

File: src/ServicePulse.Host/vue/src/components/failedmessages/AllFailedMessages.vue
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/AllFailedMessages.vue:1]: same as before `loadMessages` not awaited

**Review by johnsimons — CHANGES_REQUESTED** [review]
I added the following eslint rule, and there were some errors @mikeminutillo 
image

File: src/ServicePulse.Host/vue/src/components/failedmessages/LastTenOperations.vue [hunk 1]
```diff
@@ -7,15 +7,12 @@ import TimeSince from "../TimeSince.vue";
...
-function getHistoricOperations() {
-  return useFetchFromServiceControl("recoverability/history")
-    .then((response) => {
-      return response.json();
-    })
-    .then((data) => {
-      historicOperations.value = [];
-      historicOperations.value = data.historic_operations;
-    });
+async function getHistoricOperations() {
```
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/LastTenOperations.vue:10]: not awaited

**Review by johnsimons — CHANGES_REQUESTED** [review]
I decided to install https://github.com/SebastienGllmt/eslint-plugin-no-floating-promise. This is one of the rules we need to enforce the proper `await/async`.
After enabling it and running the lint, it seems we have a few missing awaits!
image

**Review by johnsimons — APPROVED** [review]

**Review by PhilBastian — APPROVED** [review]

**Review by johnsimons — APPROVED** [review]

**Pull request merged**